src/Sylius/Bundle/ApiBundle/spec/Serializer/ProductVariantNormalizerSpec.php

```diff
@@ -152,10 +156,13 @@ function it_returns_catalog_promotions_if_applied(
 
         $variant->getAppliedPromotionsForChannel($channel)->willReturn(new ArrayCollection([$catalogPromotion->getWrappedObject()]));
         $availabilityChecker->isStockAvailable($variant)->willReturn(true);
-        $iriConverter->getIriFromResource($catalogPromotion)->willReturn('/api/v2/shop/catalog-promotions/winter_sale');
+        $iriConverter
+            ->getIriFromResource($catalogPromotion, UrlGeneratorInterface::ABS_PATH, null, array_merge($context, [self::ALREADY_CALLED => true]))
```

Comment:
I would be for an explicit passing of the context instead of setting the variable
```suggestion
            ->getIriFromResource($catalogPromotion, UrlGeneratorInterface::ABS_PATH, null, [ContextKeys::CHANNEL => $channel, self::ALREADY_CALLED => true]))
```

Reply:
The `ProductVariantNormalizer` adds to the `context` variable `self::ALREADY_CALLED => true`, so I need to have `context` that I will pass to the function and one that will be passed to the iriConverter.